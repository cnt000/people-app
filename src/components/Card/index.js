import React from 'react'
import styled, {css} from 'styled-components';

const Div = styled.div`
  padding: 0.25em 1em;
  width: 10em;
  height: 16em;
  border: 2px solid black;
  color: black;
  border-radius: 0.25em;
  background-color: lightgray;
  margin: 0.2em;
  font-size: 16px;

  ${props => props.showed && css`
    background: green;
  `}
`;

class Card extends React.Component {

  constructor(props) {
    super(props)
    this.state = { showed: this.props.showed }
  }

  show() {
    this.setState({ showed: true })
  }

  render() {
    return <Div 
      showed={this.state.showed}
      onClick={() => this.show()}>
        {this.state.showed && this.props.value}
      </Div>
  }
}

export default Card