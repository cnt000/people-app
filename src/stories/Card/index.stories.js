import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Card from "../../components/Card"

storiesOf('Card', module)
  .add('default', () => <Card value="A" showed={false} onClick={action('clicked')} />)
  .add('showed', () => <Card value="B" showed={true} onClick={action('clicked')} />)
