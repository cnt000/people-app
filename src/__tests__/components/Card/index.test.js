import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../../../components/Card';
import { shallow } from 'enzyme';

it('Card render', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card value={"A"} showed={false} onClick={() => console.log('click')} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Card render showed', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card value={"A"} showed={true} onClick={() => console.log('click')} />, div);
  ReactDOM.unmountComponentAtNode(div);
});


// it('Card renders showed', () => {
//   shallow(<Card value={"A"} showed={true} onClick={() => console.log('click')} />);
// });
