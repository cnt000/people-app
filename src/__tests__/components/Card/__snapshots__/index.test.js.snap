// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card fire an onClick 1`] = `
<styled.div
  onClick={[Function]}
  showed={true}
>
  A
</styled.div>
`;

exports[`Card should render correctly hided 1`] = `
<styled.div
  onClick={[Function]}
  showed={false}
/>
`;

exports[`Card should render correctly showed 1`] = `
<styled.div
  onClick={[Function]}
  showed={true}
>
  A
</styled.div>
`;
